Example/PaymentSheet Example/PaymentSheet Example/ExampleCustomDeferredCheckoutViewController.swift

```diff
@@ -0,0 +1,334 @@
+    @IBOutlet weak var hotDogStepper: UIStepper!
+    @IBOutlet weak var saladStepper: UIStepper!
+    @IBOutlet weak var subtotalLabel: UILabel!
+    @IBOutlet weak var salesTaxLabel: UILabel!
+    @IBOutlet weak var totalLabel: UILabel!
+    @IBOutlet weak var subscribeSwitch: UISwitch!
+
+    private var paymentSheetFlowController: PaymentSheet.FlowController!
+    private let baseUrl = "https://stripe-mobile-payment-sheet-custom-deferred.glitch.me"
+    private var backendCheckoutUrl: URL {
+        return URL(string: baseUrl + "/checkout")!
+    }
+    private var createIntentUrl: URL {
+        return URL(string: baseUrl + "/create_intent")!
+    }
+    private var computeTotalsUrl: URL {
+        return URL(string: baseUrl + "/compute_totals")!
+    }
+
+    private struct ComputedTotals: Decodable {
+        let subtotal: Double
+        let tax: Double
+        let total: Double
+    }
+
+    private var computedTotals: ComputedTotals!
+
+    private var intentConfig: PaymentSheet.IntentConfiguration {
+        return .init(mode: .payment(amount: Int(computedTotals.total * 100),
+                                    currency: "USD",
+                                    setupFutureUsage: subscribeSwitch.isOn ? .offSession : nil),
+                     confirmHandler: confirmHandler(_:_:))
+    }
+
+    private var currencyFormatter: NumberFormatter {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        formatter.maximumFractionDigits = 2
+        return formatter
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        buyButton.addTarget(self, action: #selector(didTapCheckoutButton), for: .touchUpInside)
+        buyButton.isEnabled = false
+
+        paymentMethodButton.addTarget(self, action: #selector(didTapPaymentMethodButton), for: .touchUpInside)
+        paymentMethodButton.isEnabled = false
+
+        hotDogStepper.isEnabled = false
+        saladStepper.isEnabled = false
+        subscribeSwitch.isEnabled = false
+
+        self.loadCheckout()
+    }
+
+    // MARK: - Button handlers
+
+    @objc
+    func didTapPaymentMethodButton() {
+        // MARK: Present payment options to the customer
+        paymentSheetFlowController.presentPaymentOptions(from: self) {
+            self.updateButtons()
+        }
+    }
+
+    @objc
+    func didTapCheckoutButton() {
+        // MARK: Confirm payment
+        paymentSheetFlowController.confirm(from: self) { paymentResult in
+            // MARK: Handle the payment result
+            switch paymentResult {
+            case .completed:
+                self.displayAlert("Your order is confirmed!")
+            case .canceled:
+                print("Canceled!")
+            case .failed(let error):
+                print(error)
+                self.displayAlert("Payment failed: \n\(error.localizedDescription)")
+            }
+        }
+    }
+
+    @IBAction func hotDogStepperDidChange() {
```

Comment:
I'm not sure if this is possible but can we point these UIs directly to the updateUI function instead of having these middlemen funcs?

Reply:
Yeah I tried to consolidate but interface builder seems to not support this.